digdag-standards/src/main/java/io/digdag/standards/operator/td/TdOperatorFactory.java

```diff
@@ -78,46 +85,126 @@ public Operator newTaskExecutor(Path workspacePath, TaskRequest request)
+            this.resultUrl = params.getOptional("result_url", String.class);
 
-            int jobRetry = params.get("job_retry", int.class, 0);
+            this.jobRetry = params.get("job_retry", int.class, 0);
 
-            String engine = params.get("engine", String.class, "presto");
+            this.engine = params.get("engine", String.class, "presto");
 
-            Optional<String> downloadFile = params.getOptional("download_file", String.class);
+            this.downloadFile = params.getOptional("download_file", String.class);
             if (downloadFile.isPresent() && (insertInto.isPresent() || createTable.isPresent())) {
                 // query results become empty if INSERT INTO or CREATE TABLE query runs
                 throw new ConfigException("download_file is invalid if insert_into or create_table is set");
             }
 
-            boolean storeLastResults = params.get("store_last_results", boolean.class, false);
+            this.storeLastResults = params.get("store_last_results", boolean.class, false);
+
+            this.preview = params.get("preview", boolean.class, false);
+
+            this.state = request.getLastStateParams().deepCopy();
+
+            this.existingJobId = state.getOptional(JOB_ID, String.class);
+            this.existingDomainKey = state.getOptional(DOMAIN_KEY, String.class);
+        }
+
+        @Override
+        public TaskResult runTask()
+        {
+            try (TDOperator op = TDOperator.fromConfigWithRetries(params)) {
+
+                // Generate and store domain key before starting the job
+                if (!existingDomainKey.isPresent()) {
+                    String domainKey = UUID.randomUUID().toString();
+                    state.set(DOMAIN_KEY, domainKey);
+                    throw TaskExecutionException.ofNextPolling(0, ConfigElement.copyOf(state.deepCopy()));
+                }
+
+                // Start the job
+                if (!existingJobId.isPresent()) {
+                    String jobId = startJob(op);
+                    state.set(JOB_ID, jobId);
+                    state.set(POLL_INTERVAL, INITIAL_POLL_INTERVAL);
+                    throw TaskExecutionException.ofNextPolling(INITIAL_POLL_INTERVAL, ConfigElement.copyOf(state.deepCopy()));
+                }
+
+                // Check if the job is done
+                String jobId = existingJobId.get();
+                TDJobOperator job = op.newJobOperator(jobId);
+                TDJobSummary status = checkJobStatus(job);
+                boolean done = status.getStatus().isFinished();
+                if (!done) {
+                    int prevPollInterval = state.get(POLL_INTERVAL, int.class, INITIAL_POLL_INTERVAL);
+                    int nextPollInterval = Math.min(MAX_POLL_INTERVAL, prevPollInterval * 2);
+                    state.set(POLL_INTERVAL, nextPollInterval);
```

Comment:
I think it's easier to include number of retries in stateParams.
Polling interval can be calculated using `min(initialInterval * pow(2, retryCount), maxInterval)`.